pytensor/graph/basic.py

```diff
@@ -1896,9 +1896,9 @@ def equal_computations(
             if isinstance(x, Constant):
                 return np.array_equal(x.data, y)
             return False
-        if x.owner and not y.owner:
+        if x.owner and y.owner is None:
             return False
-        if y.owner and not x.owner:
+        if y.owner and x.owner is None:
```

Comment:
In this case the if `x.owner is not None and ...` seems fine. Maybe also all, anyway see my reasoning in the main thread.